### DISCUSSION TITLE
template massage

### DISCUSSION THREAD

**Discussion by ridone713** [filamentphp/filament#12050] [Help]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

Tables\Actions\Action::make('wa')
                    ->label(false)
                    ->color('success')
                    ->button()->link()->icon('heroicon-o-chat-bubble-left-ellipsis')
                    ->size(ActionSize::Medium)
                    ->url(fn (Tagihan $record): string => "https://wa.me/{$record->no_hp}")
                    ->openUrlInNewTab(),

Here's the translation:

"How can we directly contact the target WhatsApp number, but the WhatsApp number column is in the Penghuni table, and then add a template message directly to the target number?"

**Comment by leandrocfe** [comment]
Penghuni table, what does it mean? Do you have a relationship?

**Reply by ridone713** [reply]
yes sir

**Reply by ridone713** [reply]
"I want to execute a record by pressing an action button in the tagihan table to directly connect to the WhatsApp number(no_hp), but the no_hp column is located in the 'penghuni' table."

**Reply by leandrocfe** [reply]
`->url(fn ($record): string => "https://wa.me/{$record->tagihan->no_hp}")`

?

**Reply by ridone713** [reply]
it says: 
Attempt to read property "no_hp" on int

**Reply by leandrocfe** [reply]
tagihan isn't a relationship, I guess

**Reply by ridone713** [reply]
"Okay, finally succeeded, sir. But how do I add a template message to that WhatsApp number?"

**Reply by ridone713** [reply]
from this: 
![wa1](https://github.com/filamentphp/filament/assets/164133045/db5a156d-7663-425d-b5b6-82a465c8f01c)

to this:
![image](https://github.com/filamentphp/filament/assets/164133045/1ca52bdd-3f09-4090-a339-890d51a048cf)
